src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -373,88 +375,30 @@ public void onWorldTick(TickEvent.WorldTickEvent event)
-                {
-                    if (!event.world.isBlockLoaded(new BlockPos(getCenter().getX() + x, 128, getCenter().getZ() + z)))
-                    {
-                        allColonyChunksLoaded = false;
-                        break;
-                    }
-                }
-
-                if (!allColonyChunksLoaded)
-                {
-                    break;
-                }
-            }
-
-            if (allColonyChunksLoaded)
+            if ((event.world.getWorldTime() % CITIZEN_CLEANUP_TICK_INCREMENT) == 0 && areAllColonyChunksLoaded(event))
             {
                 //  All chunks within a good range of the colony should be loaded, so all citizens should be loaded
                 //  If we don't have any references to them, destroy the citizen
-
-                citizens.values().stream().filter(citizen -> citizen.getCitizenEntity() == null).forEach(citizen -> {
-                    Log.logger.warn(String.format("Citizen #%d:%d has gone AWOL, respawning them!", getID(), citizen.getId()));
-                    spawnCitizen(citizen);
-                });
-            }
-        }
-
-        //  Cleanup Buildings whose Blocks have gone AWOL
-        if (event.phase == TickEvent.Phase.START)
-        {
-            List<Building> removedBuildings = null;
-
-            for (Building building : buildings.values())
-            {
-                BlockPos loc = building.getLocation();
-                if (event.world.isBlockLoaded(loc) &&
-                        !building.isMatchingBlock(event.world.getBlockState(loc).getBlock()))
-                {
-                    //  Sanity cleanup
-                    if (removedBuildings == null)
-                    {
-                        removedBuildings = new ArrayList<>();
-                    }
-                    removedBuildings.add(building);
-                }
+                citizens.values().stream().filter(citizen -> citizen.getCitizenEntity() == null)
+                        .forEach(citizen -> {
+                            Log.logger.warn(String.format("Citizen #%d:%d has gone AWOL, respawning them!", getID(), citizen.getId()));
+                            spawnCitizen(citizen);
+                        });
             }
 
-            if (removedBuildings != null)
-            {
-                removedBuildings.forEach(Building::destroy);
-            }
-        }
+            //  Cleanup Buildings whose Blocks have gone AWOL
+            cleanUpBuildings(event);
 
-        //  Spawn Citizens
-        if (event.phase == TickEvent.Phase.START &&
-                townHall != null)
-        {
-            if (citizens.size() < maxCitizens)
+            //  Spawn Citizens
+            if (townHall != null && citizens.size() < maxCitizens)
             {
                 int respawnInterval = Configurations.citizenRespawnInterval * 20;
                 respawnInterval -= (60 * townHall.getBuildingLevel());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)